include/caf/caf_overview.rst

```diff
@@ -0,0 +1,182 @@
+Using CAF allows you to have a consistent event-based architecture in your application.
+
+One of the applications that are using the CAF modules in |NCS| is :ref:`nrf_desktop`.
+
+In an event-based application, parts of the application functionality are separated into isolated modules that communicate with each other using Event Manager's events.
+These events are defined either by CAF or by the application.
+They can be submitted by modules and other modules can subscribe and react to them.
+
+.. figure:: ../../doc/nrf/images/caf_overview.svg
+   :alt: Common Application Framework architecture overview
+
+   Common Application Framework architecture overview
+
+Every CAF module can use additional libraries or drivers.
+The drivers can be used to communicate with hardware.
+The module can then broadcast information from the driver using Event Manager's events (for example, submit a :c:struct:`button_event` on a button press).
+It can also forward information from application to the driver (for example, display a given LED effect upon receiving :c:struct:`led_event`).
+
+In a user application, you can use any number of CAF modules with your own custom modules, and they can use both CAF and custom events.
+The figure shows an example application that uses one CAF module and one custom module, which has one custom event.
+
+Enabling CAF
+************
+
+To enable CAF, set the :option:`CONFIG_CAF` Kconfig option, initialize Event Manager, and submit the first :c:struct:`module_state_event`.
+You can then enable each :ref:`CAF module <lib_caf>` separately.
+
+Initializing Event Manager
+==========================
+
+As CAF uses Event Manager, you must initialize Event Manager in your :file:`main.c` file.
+To do so, complete the following steps:
+
+1. Include :file:`event_manager.h`.
+#. Call :c:func:`event_manager_init()`.
+
+Submitting ``module_state_event``
+=================================
+
+CAF modules are initialized when they receive a :c:struct:`module_state_event` that reports ``MODULE_STATE_READY`` of the module ``main``.
+For this to happen, you need to do the following:
+
+1. Add the following in your :file:`main.c` file:
+
+.. code-block:: c
+
+   #define MODULE main
+   #include <caf/events/module_state_event.h>
+
+#. Call the following function after Event Manager is initialized:
+
+.. code-block:: c
+
+   module_set_state(MODULE_STATE_READY);
+
+This function call submits the required :c:struct:`module_state_event`.
+For an example implementation, see :file:`/applications/nrf_desktop/src/main.c`.
+
+CAF modules
+***********
+
+CAF supports a set of :ref:`predefined modules <lib_caf>`.
+These modules are available for use out of the box.
+You only need to enable and configure them.
+
+There is no limit to the number of modules and events that can be used.
+However, Event Manager sets the limit of event types in an application to 32, and CAF inherits this limitation.
```

Comment:
@MarekPieta where this limitation comes from? Enabling bit-mask?

Reply:
The bitmask that determines if event is displayed in logs (`event_manager_displayed_events`) is of type `uint32_t`.  We assumed that this number of events will be enough for an application. In case we need more, we could switch to using an array here.